File: src/theSchitzoLog/js/databaseAccess.js
const {Client} = require('pg')

//var loginLink = fs.readFileSync("").toString();
var client = new Client("postgresql://postgres:123@localhost:5432/postgres")
var currentPlaylist = "playlist1"

/*
fetch("link1.txt")
.then(res => res.text)
.then(text => console.log(text))
*/
///////////////////////////////////////////////////////////////
//    ______                    _    _                       //
//   |  ____|                  | |  (_)                      //
//   | |__  _   _  _ __    ___ | |_  _   ___   _ __   ___    //
//   |  __|| | | || '_ \  / __|| __|| | / _ \ | '_ \ / __|   //
//   | |   | |_| || | | || (__ | |_ | || (_) || | | |\__ \   //
//   |_|    \__,_||_| |_| \___| \__||_| \___/ |_| |_||___/   //
//                                                           //
///////////////////////////////////////////////////////////////

function errorMessage(err){
    console.log(`Something Fucked Up\n\n${err}`)
}

let out = ``;
function getData() {
    client.connect()
    .then(() => client.query (`select * from ${currentPlaylist}`))
    .then(data => {for(let i = 0; i < data.rowCount; i++) {
        out += `
            <tr>
                <td>${data.rows[i].id}</td>
                <td>${data.rows[i].song_name}</td>
                <td>${data.rows[i].artist}</td>
            </tr>
            `
        }
    })
//    .then(() => document.getElementById("tableData").innerHTML = out)
    .then(() => console.log(out))
    .catch(err => errorMessage(err))
    .finally(() => client.end())
}

/////////////////////////////////////////////
//     _____              _         _      //
//    / ____|            (_)       | |     //
//   | (___    ___  _ __  _  _ __  | |_    //
//    \___ \  / __|| '__|| || '_ \ | __|   //
//    ____) || (__ | |   | || |_) || |_    //
//   |_____/  \___||_|   |_|| .__/  \__|   //
//                          | |            //
//                          |_|            //
/////////////////////////////////////////////

getData()

File: src/theSchitzoLog/py/databaseAccess.py
import pg8000 as pg

currentPlaylist = "playlist1"

####################################################
#    _____        _        _                       #
#   |  __ \      | |      | |                      #
#   | |  | | __ _| |_ __ _| |__   __ _ ___  ___    #
#   | |  | |/ _` | __/ _` | '_ \ / _` / __|/ _ \   #
#   | |__| | (_| | || (_| | |_) | (_| \__ \  __/   #
#   |_____/ \__,_|\__\__,_|_.__/ \__,_|___/\___|   #
#                                                  #
####################################################  

# with open('./py/loginInfo.txt', "r") as textInfo:
#     loginInfo = textInfo.readlines()
#     for a in enumerate(loginInfo):
#         loginInfo[a[0]] = loginInfo[a[0]].replace("\n", "")

client = pg.connect(
   database = "mhrahked",
   user = "mhrahked", 
   password = "SQAsoEhXFmHwC3Z8jlZWGpBr2WO2gmwz",
   host = "kashin.db.elephantsql.com",
   port = 5432
).cursor()

# client = pg.connect(
#    database = loginInfo[0],
#    user = loginInfo[1], 
#    password = loginInfo[2],
#    host = loginInfo[3],
#    port = loginInfo[4]
# ).cursor()

#############################################################
#    ______                    _    _                       #
#   |  ____|                  | |  (_)                      #
#   | |__  _   _  _ __    ___ | |_  _   ___   _ __   ___    #
#   |  __|| | | || '_ \  / __|| __|| | / _ \ | '_ \ / __|   #
#   | |   | |_| || | | || (__ | |_ | || (_) || | | |\__ \   #
#   |_|    \__,_||_| |_| \___| \__||_| \___/ |_| |_||___/   #
#                                                           #
#############################################################


def errorMessage(err:str):
    if len(err) == 0:
        print(f"Something Fucked Up\n\nEmpty Error")
        exit
    else:
        print(f"Something Fucked Up\n\n{err}")
        exit


def getData(currentPlaylist:str) -> list:
    client.execute(f"SELECT * FROM {currentPlaylist}")
    query = client.fetchall()
    data = []
    for a in query:
        data += [{
            'id': f'{a[0]}',
            'songName': f'{a[1]}',
            'artist': f'{a[2]}'
        }]
    return data

def putIntoTable(data:list) -> str:
    out = str()
    for a in data:
        out = out + f"""
        <tr>
            <td>{a["id"]}</td>
            <td>{a["songName"]}</td>
            <td>{a["artist"]}</td>
        </tr>""" 
    return (out)

